MHArmory.Search/OpenCL/search.cl

```diff
@@ -0,0 +1,236 @@
+#ifndef MAX_RESULTS
+	#define MAX_RESULTS 128
+#endif
+
+#ifndef MAX_DESIRED_SKILLS
+	#define MAX_DESIRED_SKILLS 0x80
+#endif
+
+#ifndef MAX_SET_SKILLS
+	#define MAX_SET_SKILLS 0x40
+#endif
+
+#ifndef MAX_JEWEL_SIZE
+	#define MAX_JEWEL_SIZE 3
+#endif
+#define SLOT_ARR_LEN MAX_JEWEL_SIZE + 1
+
+#ifndef SLOTS_PER_EQUIPMENT
+	#define SLOTS_PER_EQUIPMENT 3
+#endif
+
+#ifndef SKILLS_PER_EQUIPMENT
+	#define SKILLS_PER_EQUIPMENT 3
+#endif
+
+#ifndef SET_SKILLS_PER_EQUIPMENT
+	#define SET_SKILLS_PER_EQUIPMENT 3
+#endif
+
+#ifndef EQUIPMENT_TYPES
+	#define EQUIPMENT_TYPES 6
+#endif
+
+typedef signed char             int8_t;
```

Comment:
I usually prefer too avoid aligning names / types like this, adding a new type requires to shift everything, using another editor may keep removing whitespaces, etc...